include/exadg/fluid_structure_interaction_precice/exadg_interface.h

```diff
@@ -48,10 +46,15 @@ class ExaDGInterface : public CouplingInterface<dim, data_dim, VectorizedArrayTy
   write_data(const LinearAlgebra::distributed::Vector<double> & data_vector,
              const std::string &                                data_name) override;
 
-  virtual std::vector<Tensor<1, dim>>
+  virtual void
   read_block_data(const std::string & data_name) const override;
 
+  void
+  set_data_pointer(std::shared_ptr<ExaDG::InterfaceCoupling<dim, dim, double>> exadg_terminal_);
+
 private:
+  /// Interface for ExaDG data structures
```

Comment:
I would avoid the name "interface" here, since `InterfaceCoupling` means something like `SurfaceCoupling`.

Reply:
Replaced it by `Accessor`